src/components/DateInput/DateInput.tsx

```diff
@@ -1,57 +1,50 @@
-import React from 'react';
-import DayPickerInput from 'react-day-picker/DayPickerInput';
+import React, { useState, useCallback } from 'react';
 import dayjs from 'dayjs';
+import Box from '../Box';
+import Card from '../Card';
+import Flex from '../Flex';
+import Button from '../Button';
+import Month from './Month';
+import FadeIn from '../FadeIn';
+import { IconButton } from '../../index';
 import TextInput, { TextInputProps } from '../TextInput';
-import OverlayComponent from './OverlayComponent';
+import TimePicker from './TimePicker';
+import { noop } from '../../utils/helpers';
 
-export type DateInputProps = TextInputProps & {
+export interface DateInputProps extends TextInputProps {
   /**
-   * A string that describes the format that will be shown to the user. Under the hood, DateInput
-   * uses `dayjs` so the available values of this format can be found in its docs:
-   * https://github.com/iamkun/dayjs/blob/dev/docs/en/API-reference.md#list-of-all-available-formats
-   * */
+   * The format displayed in the input element
+   */
   format?: string;
 
   /**
-   * A string that represents the formatted Date. It will be parsed according to the given format
-   * and converted to an actual date
+   * A flag that dictates if the component should allow manipulating time
+   */
+  withTime?: boolean;
+
+  /**
+   * A date that that works as a value
    */
-  value: string;
```

Comment:
This should actually be a date **string**, since when we "boot up" the component with an existing value , the API will provide a string value.

I think this value should be the default date ISO value, which `dayjs` knows hoow to parse. Thus, when you do:

```ts
  const dateFormatted = date ? dayjs(date) : dayjs();
```

even when `date` is a "string" it will still work.

Reply:
That's intentional as well. The thing is that `Date` objects are well structured and play great along with `dayjs`. Since we need to do lots of comparisons, additions, and subtractions to the original value (we need to calculate weeks per month, hours, and minutes and eventually propagate the using the `onChange` handler), using a string values doubles the effort and the complexity.
Furthermore, the API of the component should be agnostic to the format of the value passed through. The consumers are responsible for parsing the string values per case.

Reply:
Gotcha, is there a reason then we can't name  the props `value` and `onChange` so they can be automatically handled  by libraries (all oof our inputs have the  same name, that's why)

Reply:
Fixed the API to use props `value` and `onChange`